src/send/ValidateRecipientAccount.tsx

```diff
@@ -66,19 +68,26 @@ const mapDispatchToProps = {
 const mapStateToProps = (state: RootState, ownProps: OwnProps): StateProps => {
   const { route } = ownProps
   const transactionData = route.params.transactionData
-  const { recipient } = transactionData
+  const { recipient } = route.params
   const e164PhoneNumber = recipient.e164PhoneNumber
   const error = state.alert ? state.alert.underlyingError : null
+  const secureSendPhoneNumberMapping = state.identity.secureSendPhoneNumberMapping
   const validationSuccessful = e164PhoneNumber
-    ? !!state.identity.secureSendPhoneNumberMapping[e164PhoneNumber]?.validationSuccessful
+    ? !!secureSendPhoneNumberMapping[e164PhoneNumber]?.validationSuccessful
     : false
```

Comment:
```suggestion
    && !!secureSendPhoneNumberMapping[e164PhoneNumber]?.validationSuccessful
```
code golf. you may also need !! in front of e164PhoneNumber but I wasn't able to include that in my suggestion